### ISSUE TITLE
Syncnet Probabilities

### ISSUE THREAD

**Issue by calderkatyal** [Rudrabha/Wav2Lip#744]
I'm trying to figure out how to interpret the probabilities for the syncnet model, but there appears to be a mistake (or at least anomaly) in the code. In particular, lines 133-138 of `color_syncnet_train.py` have:

```
logloss = nn.BCELoss()
def cosine_loss(a, v, y):
    d = nn.functional.cosine_similarity(a, v)
    loss = logloss(d.unsqueeze(1), y)

    return loss
```
In particular, the cosine similarity is bounded in [-1, 1] (might be negative even if the embeddings are ReLU activated), and so training could throw an error, as the input to torch.BCELoss() is expected to be positive. In particular, the value of cosine similarity tends to be very low (< .3) for the majority of examples I test it with.

What is the correct interpretation of the model probabilities? Should it be `sigmoid(cosine_similarity(a,v))` or is the intended loss function `torch.BCEWithLogitsLoss()`?

**Comment by NixNait** [comment]
The model uses the ReLU activation function, and the output feature vectors of the audio encoder and the facial image encoder are non-negative. After normalization, the calculated cosine similarity will range from 0 to 1.
